src/lib.rs

```diff
@@ -225,16 +225,21 @@ impl Navigator {
         self.bmp.pressure_kpa().unwrap()
     }
 
-    pub fn read_adc(&mut self) -> ADCData {
+    pub fn read_adc_all(&mut self) -> ADCData {
         ADCData {
             channel: [
-                block!(self.adc.read(&mut channel::SingleA0)).unwrap(),
-                block!(self.adc.read(&mut channel::SingleA1)).unwrap(),
-                block!(self.adc.read(&mut channel::SingleA2)).unwrap(),
-                block!(self.adc.read(&mut channel::SingleA3)).unwrap(),
+                block!(self.adc.read(adc_Channel::SingleA0)).unwrap(),
```

Comment:
why we need to call block in a function that already encapsulates block ?

Reply:
it works if we remove, 
it's not clear why all the examples from the library use this macro

Reply:
you are right,
linux_embedded_hal

already uses the blocking methods from embedded_hal
https://docs.rs/embedded-hal/latest/embedded_hal/blocking/

Reply:
did a bit of confusion with block macro vs the embedded_hall::block